### PR TITLE
add wrapper helpers + basic key type

### PR DESCRIPTION
**Pull request by tanner0101**
- [x] Adds `DecoderUnwrapper`

```swift
let data = "{}".data(using: .utf8)!
let unwrapper = try JSONDecoder().decode(DecoderUnwrapper.self, from: data)
print(unwrapper.decoder) // Decoder
```

- [x] Adds `EncodableWrapper`

```swift
let encodable: Encodable = ["hello": "world"]
let data = try JSONEncoder().encode(EncodableWrapper(encodable))
print(data)
```

- [x] Adds `StringCodingKey`

Basic `String` conformance to `CodingKey`.

### CODE REVIEW

File: Sources/CodableKit/DecoderUnwrapper.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/// Used to unwrap the `Decoder` from a private implementation like `JSONDecoder`.
+///
+///     let unwrapper = try JSONDecoder().decode(DecoderUnwrapper.self, from: ...)
+///     print(unwrapper.decoder) // Decoder
+///
+public struct DecoderUnwrapper: Decodable {
```
Comment (calebkleveter) [inline Sources/CodableKit/DecoderUnwrapper.swift:6]: Cool, but is this something we should really be doing in practice?

Comment (tanner0101) [reply Sources/CodableKit/DecoderUnwrapper.swift:6]: It's incredibly useful for getting raw access to Decoder methods like `container(keyedBy:)`, `singleValueContainer()`, etc. I find myself re-implementing this in most frameworks that use `Codable`.

Comment (calebkleveter) [reply Sources/CodableKit/DecoderUnwrapper.swift:6]: 👍 Then I think we should add a convenience method to get internal decoders from `DataDecoder` types.

Comment (calebkleveter) [reply Sources/CodableKit/DecoderUnwrapper.swift:6]:
```swift
JSONDecoder.decoder(with: data) // -> _JSONDecoder
```

Comment (tanner0101) [reply Sources/CodableKit/DecoderUnwrapper.swift:6]: Agreed, just put a comment on #2.

**Review by calebkleveter — APPROVED** [review]

**Comment by penny-coin** [comment]
Hey @tanner0101, you just merged a pull request, have a coin! 

You now have 1035 coins.

**Pull request merged**